### DISCUSSION TITLE
[Question]: Sharing a camera stream with someone, any ideas?

### DISCUSSION THREAD

**Discussion by DrSpaldo** [blakeblackshear/frigate#11827] [Ask A Question]
### What is your question:

I was hoping to share a camera stream from Frigate with someone. I do not want to give them access to all of the cameras nor to any of the extra features that Frigate offers, such as events/recordings/etc. 

Has anyone else done this? and how did you achieve it?

**Comment by DrSpaldo** [comment]
I've tried using the link http://192.168.1.10:5000/api/cameraname but the quality isn't great. The WebRTC links (using 1984) port won't work. 

Does the new authentication system with users allow you to restrict access for that user? ie. they can see live but not look at events/recordings?

**Reply by NickM-27** [reply]
There is no user roles yet, that will come in a future update

**Reply by DrSpaldo** [reply]
> There is no user roles yet, that will come in a future update

Thanks for letting us know. Hopefully it’s sooner than later. 

Is there a way to disable anonymous access yet? 

Have a missed one of the api links to stream a higher quality feed?

**Reply by NickM-27** [reply]
yes, just don't map port 5000, or only map it on the internal docker network

**Reply by DrSpaldo** [reply]
Sorry, I missed the info on the release notes..

_Authentication

Frigate now supports authentication with support for two different modes: native and proxy. In native mode, Frigate manages users and authentication using JWT tokens via a cookie or authorization header. Proxy mode is specifically designed to make use of common authentication proxies such as authelia, authentik, oauth2_proxy, etc.

Authentication is enforced on a new port 8080, and port 5000 is still unauthenticated for backwards compatibility. If you want to use Frigate's authentication, update your reverse proxy to point at port 8080 instead of 5000.
Support for role based authorization (admin/user roles) will come in a future release and will support upstream authentication proxies.
We are planning to include TLS support in 0.14.0 as well.
The Home Assistant integration needs to be pointed at port 5000. Some recommended configuration examples are in the [updated docs](https://deploy-preview-11419--frigate-docs.netlify.app/integrations/home-assistant#configuration)._
